### PR TITLE
Add two minor fixes for data download

### PR DESCRIPTION
**Pull request by davidotte**
### Change Description

*Try to be precise. You can additionally add comments to your PR, this might help the reviewer a lot.*

Two minor fixes:
- Client version header was missing in one request
- Add extra if statement in data validation for data download

**If you used new dependencies: Did you add them to `requirements.txt`?**

**Who did you ping on Mattermost to review your PR? Please ping that person again whenever you are ready for another review.**

@SamuelGabriel 

## Breaking changes

If you made any breaking changes, please update the version number.
Breaking changes are totally fine, we just need to make sure to keep the users informed and the server in sync.

no

**Does this PR break the API? If so, what is the corresponding server commit?**

no

**Does this PR break the user interface? If so, why?**

no

---
*Please do not mark comments/conversations as resolved unless you are the assigned reviewer. This helps maintain clarity during the review process.*

### CODE REVIEW

File: tabpfn_client/client.py [hunk 1]
```diff
@@ -193,6 +193,8 @@ def _validate_response(
         # Read response.
         load = None
         try:
+            if not response.is_closed:
```
Comment (SamuelGabriel) [inline tabpfn_client/client.py:196]: what does this do? could you add a comment why you do this?

**Review by SamuelGabriel — APPROVED** [review]
looks good generally

File: tabpfn_client/client.py (hunk 1 above)
Comment (davidotte) [reply tabpfn_client/client.py:196]: This is to make sure that the response is read before response.json() gets called. When dealing with a streaming response (like during data download), I sometimes got an httpx.ResponseNotRead error without this if block. I will add a comment.

Comment (SamuelGabriel) [reply tabpfn_client/client.py:198]: Oh interesting. Thank you!

**Pull request merged**